code/_onclick/hud/fullscreen.dm

```diff
@@ -1,125 +1,104 @@
+		// doesn't need to be updated
+		return screen
 
 	screen.icon_state = "[initial(screen.icon_state)][severity]"
 	screen.severity = severity
-
-	if (rescale)
-		screen.set_size(client)
-
-	screens[category] = screen
-	if(client && (stat != DEAD || screen.allstate))
-
+	if (client && SHOULD_SHOW_TO(src, screen))
+		screen.update_for_view(client.view)
 		client.screen += screen
+
 	return screen
 
 /mob/proc/clear_fullscreen(category, animated = 10)
-
 	var/obj/screen/fullscreen/screen = screens[category]
-
 	if(!screen)
 		return
 
 	screens -= category
 
 	if(animated)
-		spawn(0)
-			animate(screen, alpha = 0, time = animated)
-			sleep(animated)
-			if(client)
-				client.screen -= screen
-			qdel(screen)
+		animate(screen, alpha = 0, time = animated)
+		addtimer(CALLBACK(src, .proc/clear_fullscreen_after_animate, screen), animated, TIMER_CLIENT_TIME)
 	else
 		if(client)
 			client.screen -= screen
 		qdel(screen)
 
+/mob/proc/clear_fullscreen_after_animate(obj/screen/fullscreen/screen)
+	if(client)
+		client.screen -= screen
+	qdel(screen)
+
 /mob/proc/clear_fullscreens()
 	for(var/category in screens)
-		clear_fullscreen(category, 0)
+		clear_fullscreen(category)
 
 /mob/proc/hide_fullscreens()
 	if(client)
 		for(var/category in screens)
 			client.screen -= screens[category]
 
-/mob/proc/reload_fullscreen()
+/mob/proc/reload_fullscreens()
 	if(client)
+		var/obj/screen/fullscreen/screen
 		for(var/category in screens)
-			var/obj/screen/fullscreen/F = screens[category]
-			var/newtype
-			if (F)
-				newtype = F.type
-			clear_fullscreen(category, 0)
-			//client.screen -= screens[category]
-			overlay_fullscreen(category, newtype, INFINITY)
-
-
-/proc/get_or_create_fullscreen(var/view_radius)
-	var/pixels = ((view_radius*2)+1)*world.icon_size
-	var/entry_name = "[pixels]x[pixels]"
-	if (!GLOB.fullscreen_icons[entry_name])
-		//If the icons isn't made yet, make it and set it in the global list
-		GLOB.fullscreen_icons[entry_name] = rescale_icon('icons/mob/screen_full.dmi', pixels, pixels)
-	return GLOB.fullscreen_icons[entry_name] //Then return it
-
+			screen = screens[category]
+			if(SHOULD_SHOW_TO(src, screen))
+				screen.update_for_view(client.view)
+				client.screen |= screen
+			else
+				client.screen -= screen
 
 /obj/screen/fullscreen
-	icon = 'icons/mob/screen_full.dmi'
+	icon = 'icons/hud/screen_full.dmi'
 	icon_state = "default"
 	screen_loc = "BOTTOMLEFT"
 	plane = FULLSCREEN_PLANE
 	mouse_opacity = 0
-	var/small_icon = FALSE	//True on any that don't use screen_full.dmi
 	var/severity = 0
-	var/allstate = 0 //shows if it should show up for dead people too
-	var/mob/owner
-
-/obj/screen/fullscreen/proc/set_size(var/client/C)
-	//Here we select (and if needed, generate) the icon for the right size
-	if (C.temp_view == world.view)
-		return	//No special sizing needed
-
-	icon = get_or_create_fullscreen(C.temp_view)
-
+	var/fs_view = WORLD_VIEW
```

Comment:
no abbreviations please, give this variable a full title descriptive of what it does

Reply:
> no abbreviations please, give this variable a full title descriptive of what it does

fullscreen_view = 480x480 (15x15 tiles)

Reply:
don't tell me, change its name to that!